From stable+bounces-119945-greg=kroah.com@vger.kernel.org Fri Feb 28 14:57:55 2025
From: Tomas Glozar <tglozar@redhat.com>
Date: Fri, 28 Feb 2025 14:57:05 +0100
Subject: Revert "rtla/timerlat_top: Set OSNOISE_WORKLOAD for kernel threads"
To: stable@vger.kernel.org
Cc: Steven Rostedt <rostedt@goodmis.org>, Luis Goncalves <lgoncalv@redhat.com>, Greg Kroah-Hartman <gregkh@linuxfoundation.org>, Guillaume Morin <guillaume@morinfr.org>, Wang Yugui <wangyugui@e16-tech.com>, Jan Kundrat <jan.kundrat@cesnet.cz>, Tomas Glozar <tglozar@redhat.com>
Message-ID: <20250228135708.604410-2-tglozar@redhat.com>

From: Tomas Glozar <tglozar@redhat.com>

This reverts commit 41955b6c268154f81e34f9b61cf8156eec0730c0.

The commit breaks rtla build, since params->kernel_workload is not
present on 6.6-stable.

Signed-off-by: Tomas Glozar <tglozar@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 tools/tracing/rtla/src/timerlat_top.c |   15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

--- a/tools/tracing/rtla/src/timerlat_top.c
+++ b/tools/tracing/rtla/src/timerlat_top.c
@@ -679,15 +679,12 @@ timerlat_top_apply_config(struct osnoise
 		auto_house_keeping(&params->monitored_cpus);
 	}
 
-	/*
-	* Set workload according to type of thread if the kernel supports it.
-	* On kernels without support, user threads will have already failed
-	* on missing timerlat_fd, and kernel threads do not need it.
-	*/
-	retval = osnoise_set_workload(top->context, params->kernel_workload);
-	if (retval < -1) {
-		err_msg("Failed to set OSNOISE_WORKLOAD option\n");
-		goto out_err;
+	if (params->user_top) {
+		retval = osnoise_set_workload(top->context, 0);
+		if (retval) {
+			err_msg("Failed to set OSNOISE_WORKLOAD option\n");
+			goto out_err;
+		}
 	}
 
 	return 0;
